4985_CommAudio/server.cpp

```diff
@@ -466,6 +477,90 @@ void CALLBACK clientRoutine(DWORD error, DWORD, LPWSAOVERLAPPED, DWORD)
 }
 
 /*--------------------------------------------------------------------------------------
+--  INTERFACE:     void CALLBACK parseRoutine(DWORD error, DWORD bytesTransferred, LPWSAOVERLAPPED overlapped, DWORD)
+--                     DWORD error: error that occured during WSARecv()
+--                     DWORD bytesTransferred: the number of bytes received
+--                     LPWSAOVERLAPPED overlapped: pointer to overlapped struct
+--                     Other arg unused
+--
+--  RETURNS:       void
+--
+--  DATE:          April 7, 2017
+--
+--  DESIGNER:      Matt Goerwell
+--
+--  PROGRAMMER:    Matt Goerwell
+--
+--  NOTES:
+--      Completion routine for receiving a client request.  Checks for errors, then parses the
+--      request type made and proceeds to the appropriate method.
+---------------------------------------------------------------------------------------*/
+void CALLBACK parseRoutine(DWORD error, DWORD bytesTransferred, LPWSAOVERLAPPED overlapped, DWORD)
+{
+    SocketInformation *si = (SocketInformation *)overlapped;
+    // Check for error or close connection request
+    if (error != 0 || bytesTransferred == 0)
+    {
+        if (error)
+        {
+            fprintf(stderr, "Error: %d\n", error);
+        }
+        fprintf(stderr, "Closing socket: %d\n", (int)si->socket);
+        closesocket(si->socket);
+        return;
+    }
+    if (strcmp("pick",si->buffer) == 0) {
+        selectSong(overlapped);
```

Comment:
Don't understand why you don't just have `selectSong` take in a pointer to SocketInformation and then pass in si.